### PR TITLE
Cart Endpoints

### PR DESCRIPTION
**Pull request by jbytes1027**

### CODE REVIEW

File: ShopAPI/ShopAPI/Services/CartService.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
+using Microsoft.EntityFrameworkCore;
+using ShopAPI.Data;
+using ShopAPI.Models;
+
+namespace ShopAPI.Services;
+
+public class CartService
+{
+    private readonly ApplicationDbContext _context;
+
```
Comment (evan-scales) [inline ShopAPI/ShopAPI/Services/CartService.cs]: Lets create an interface for cartservice as well

File: ShopAPI/ShopAPI/Services/CartService.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        return Task.FromResult(returnedItem);
+    }
+
+
+    public Task<List<CartItem>> getAllItemsAsync()
+    {
+        var cartItems = _context.CartItems.ToList();
+        return Task.FromResult(cartItems);
+    }
```
Comment (evan-scales) [inline ShopAPI/ShopAPI/Services/CartService.cs]: Pascal case please

File: ShopAPI/ShopAPI/Services/CartService.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    public Task<List<CartItem>> getCartItemsAsync(int cartId)
+    {
+        var cartItems = _context.CartItems
+            .Where(item => item.CartId == cartId)
+            .Include(item => item.Product)
+            .ToList();
+        return Task.FromResult(cartItems);
+    }
```
Comment (evan-scales) [inline ShopAPI/ShopAPI/Services/CartService.cs]: Pascal case please

File: ShopAPI/ShopAPI/Controllers/ShopController.cs [hunk 4]
```diff
@@ -119,5 +121,60 @@ public async Task<IActionResult> DeleteProduct(int id)
...
+            List<CartItem> cartItems = await _cartService.getCartItemsAsync(cartId);
+            var totals = Calculate.Totals(cartItems);
+
+            // Convert cartItems to cartItemsDTOs
+            List<CartItemDTO> cartItemDTOs = new();
+            foreach (var cartItem in cartItems)
+            {
+                CartItemDTO cartItemDTO = new(cartItem.Product.ModelToDTO(), cartItem.Quantity);
+                cartItemDTOs.Add(cartItemDTO);
+            }
```
Comment (evan-scales) [inline ShopAPI/ShopAPI/Controllers/ShopController.cs]: Interesting article I read recently that might apply here - https://www.code4it.dev/csharptips/initialize-collection-size/

Up to you if you though if you want to make the change

**Review by evan-scales — CHANGES_REQUESTED** [review]
Works fine on my end, just some small things to change

File: ShopAPI/ShopAPI/Controllers/ShopController.cs (hunk 4 above)
Comment (jbytes1027) [reply ShopAPI/ShopAPI/Controllers/ShopController.cs]: Fixed in https://github.com/evan-scales/PalmetoProgrammers547/pull/10/commits/4030628e32dc8d38ed6f61ad652d4eda96a3185b

File: ShopAPI/ShopAPI/Services/CartService.cs (hunk 2 above)
Comment (jbytes1027) [reply ShopAPI/ShopAPI/Services/CartService.cs]: Fixed in e0a9b1b16132ce743b78f5088a483018d7e35791

File: ShopAPI/ShopAPI/Services/CartService.cs (hunk 3 above)
Comment (jbytes1027) [reply ShopAPI/ShopAPI/Services/CartService.cs]: Fixed in e0a9b1b16132ce743b78f5088a483018d7e35791

**Review by evan-scales — APPROVED** [review]
good

**Pull request merged**